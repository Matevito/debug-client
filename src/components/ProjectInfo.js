import React, { useState, useEffect } from 'react'
import { Link } from "react-router-dom";
import { VictoryPie, VictoryTheme } from "victory"

// mui comp
import {
    Box,
    Typography,
    Stack
} from "@mui/material"
import PersonIcon from '@mui/icons-material/Person';

export const ProjectInfo = ({ projectData }) => {
    const [project, setProject] = useState(null)
    const [issues, setIssues] = useState([])
    const [data, setData] = useState([])

    useEffect(() => {
        if (projectData) {
            setProject(projectData.data);
            setIssues(projectData.issues);
            const openTicket = projectData.issues.filter((issue) => issue.status === "open");
            const infoNedTicket =  projectData.issues.filter((issue) => issue.status === "aditional info needed");
            const progressTicket =  projectData.issues.filter((issue) => issue.status === "in progress");
            const underReTicket =  projectData.issues.filter((issue) => issue.status === "under review");
            const solvedTicket = projectData.issues.filter((issue) => issue.status === "solved");
            let dataPrototype = [
                {label: "open", x: 1, y:openTicket.length},
                {label: "info needed", x: 2, y:infoNedTicket.length},
                {label: "in progress", x: 3, y:progressTicket.length},
                {label: "under review", x: 4, y:underReTicket.length},
                {label: "solved", x: 5, y:solvedTicket.length},
            ]
            dataPrototype = dataPrototype.filter((data) => data.y > 0)
            setData(dataPrototype)
        }
    }, [projectData])

    if (!project){
        return (
            <></>
        )
    } else  {
        return (
            
            <>
            <Box
                sx={{
                    m:2,
                    display: "flex",
                    flexDirection: "column",
                }}
            >
                <Typography variant="h4"><b>{project.title}</b></Typography>
                <Typography variant="h6" ><b>Description</b></Typography>
                <Typography component={'span'}>{project.description}</Typography>

                <Typography variant="h6"><b>Team Leader</b></Typography>
                <Typography  component={'span'}>
                    <Link to={`/user/${project.teamLeader._id}}`} style={{ color: "inherit", textDecoration: "inherit"}} >
                        <Stack direction="row" alignItems="center" gap={1}>
                            <PersonIcon color="error"/>
                            {project.teamLeader.username}
                        </Stack>
                    </Link>
                </Typography>

                <Typography variant="h6"><b>Team members</b></Typography>
                {project.team.filter(user => user._id !== project.teamLeader._id).map((user) => {
                    return(
                        <Typography key={user._id} component={'span'}>
                            <Link to={`/user/${user._id}`} style={{ color: "inherit", textDecoration: "inherit" }}>
                                <Stack direction="row" alignItems="center" gap={1}>
                                    <PersonIcon color="success"/>
                                    {user.username}
                                </Stack>
                            </Link>
                        </Typography>
                    )
                })}
                <Typography variant="h6"><b>Total tickets: </b>{issues.length}</Typography>
                {issues.length ? 
                    <VictoryPie 
                        data={data}
                        innerRadius={50}
                        theme={VictoryTheme.material}
                    /> : 
                    <></>
                }
            </Box>
            </>
        )
    }
    
}
