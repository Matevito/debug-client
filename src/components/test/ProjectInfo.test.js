import React from "react"
import { MemoryRouter } from "react-router-dom"

// testing com
import { render} from "@testing-library/react";
import '@testing-library/jest-dom'

// test comp
import { ProjectInfo } from "../ProjectInfo"

describe("ProjectInfo component", () => {
    test("handles no props", () => {
        const { container } = render(<ProjectInfo projectData={null}/>)
        expect(container).toMatchSnapshot();
    });
    test("renders basic proj components", () => {
        const projData = {
            data:{
                _id: "testProject1",
                title: "test title",
                description: "a description",
                team: [
                    {_id: "one", username:"user1"},
                    {_id:"two", username:"user2"}
                ],
                teamLeader: {
                    _id: "one", username:"user1"
                }
            },
            issues: [
                {
                    _id: "1",
                    title: "issue1",
                    priority: "low",
                    description:" a simple description",
                    handlingTeam: [],
                    status: "open",
                    date: "2022-04-23T22:56:09.627Z",
                    screenshots: []
                },
                {
                    _id: "2",
                    title: "issue2",
                    priority: "low",
                    description: "a more complicated descirption",
                    handlingTeam: [{_id:"two", username:"user2"}],
                    status: "in progress",
                    date: "2022-04-23T22:56:09.627Z",
                    screenshots: []
                }
            ]
        }
        const { container } = render(<ProjectInfo projectData={projData} />, {wrapper: MemoryRouter})
        expect(container).toMatchSnapshot();
    });
})