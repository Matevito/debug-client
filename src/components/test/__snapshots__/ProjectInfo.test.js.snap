// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectInfo component handles no props 1`] = `<div />`;

exports[`ProjectInfo component renders basic proj components 1`] = `
<div>
  <div
    class="MuiBox-root css-1nn3c3o"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      <b>
        test title
      </b>
    </h4>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      <b>
        Description
      </b>
    </h6>
    <span
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      a description
    </span>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      <b>
        Team Leader
      </b>
    </h6>
    <span
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <a
        href="/user/one}"
        style="text-decoration: inherit;"
      >
        <div
          class="css-1rlitzi-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
          user1
        </div>
      </a>
    </span>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      <b>
        Team members
      </b>
    </h6>
    <span
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <a
        href="/user/two"
        style="text-decoration: inherit;"
      >
        <div
          class="css-1rlitzi-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-4b67si-MuiSvgIcon-root"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
          user2
        </div>
      </a>
    </span>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      <b>
        Total tickets: 
      </b>
      2
    </h6>
    <div
      class="VictoryContainer"
      style="height: 100%; width: 100%; pointer-events: none; position: relative;"
    >
      <svg
        height="350"
        role="img"
        style="pointer-events: all; width: 100%; height: 100%;"
        viewBox="0 0 350 350"
        width="350"
      >
        <g>
          <path
            d="M7.654042494670958e-15,-125A125,125,0,1,1,7.654042494670958e-15,125L3.061616997868383e-15,50A50,50,0,1,0,3.061616997868383e-15,-50Z"
            role="presentation"
            shape-rendering="auto"
            style="fill: #F4511E; padding: 8px; stroke: #ECEFF1; stroke-width: 1;"
            transform="translate(175, 175)"
          />
          <path
            d="M7.654042494670958e-15,125A125,125,0,1,1,-2.2962127484012872e-14,-125L-9.184850993605149e-15,-50A50,50,0,1,0,3.061616997868383e-15,50Z"
            role="presentation"
            shape-rendering="auto"
            style="fill: #FFF59D; padding: 8px; stroke: #ECEFF1; stroke-width: 1;"
            transform="translate(175, 175)"
          />
          <text
            direction="inherit"
            dx="0"
            id="pie-labels-0"
            x="320"
            y="179.26"
          >
            <tspan
              dx="0"
              dy="0"
              style="padding: 20px; font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; fill: #455A64; stroke: transparent; stroke-width: 0;"
              text-anchor="start"
              x="320"
            >
              open
            </tspan>
          </text>
          <text
            direction="inherit"
            dx="0"
            id="pie-labels-1"
            x="30"
            y="179.26"
          >
            <tspan
              dx="0"
              dy="0"
              style="padding: 20px; font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; fill: #455A64; stroke: transparent; stroke-width: 0;"
              text-anchor="end"
              x="30"
            >
              in progress
            </tspan>
          </text>
        </g>
      </svg>
      <div
        style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
      >
        <svg
          height="350"
          style="overflow: visible; width: 100%; height: 100%;"
          viewBox="0 0 350 350"
          width="350"
        />
      </div>
    </div>
  </div>
</div>
`;
